scripts/bert/finetune_squad.py

```diff
@@ -335,28 +360,34 @@ def train():
     log.info('The number of examples after preprocessing:{}'.format(
         len(train_data_transform)))
 
+    sampler = nlp.data.SplitSampler(len(train_data_transform), num_parts=size,
+                                    part_index=rank, even_size=True)
```

Comment:
`even_size=True`? It is possible that the samples cannot be evenly split.

Reply:
```
    even_size: bool, default False
      If the number of samples is not even across all partitions, sample a few extra samples
      for the ones with fewer samples.
```

Reply:
Then, it is possible that some workers do not have gradient to contribute at the end of each epoch.

Reply:
It means that the sampler will duplicate with extra samples if the size is not even

Reply:
ok, i just saw the `extras` in the `__iter__`.